sqlglot/optimizer/annotate_types.py

```diff
@@ -235,6 +236,20 @@ def _set_type(
         elif prev_type and t.cast(exp.DataType, prev_type).this == exp.DataType.Type.NULL:
             self._null_expressions.pop(expression_id, None)
 
+        if (
+            isinstance(expression, exp.Column)
+            and expression.is_type(exp.DataType.Type.JSON)
+            and (dot_parts := expression.meta.get("dot_parts"))
+        ):
```

Comment:
@VaggelisD it just occurred to me: what if you have `FUNC_RETURNING_JSON().SomeKey`?

Reply:
Can we leave a todo and tackle it later?